apps/src/code-studio/initApp/project.js

```diff
@@ -849,13 +849,19 @@ var projects = module.exports = {
       current.projectType = this.getStandaloneApp();
     }
     this.setName(newName);
-    channels.create(current, function (err, data) {
-      this.updateCurrentData_(err, data, options);
-      this.save(
-        false /* forceNewVersion */,
-        true /* preparingRemix */
-      ).then(callback);
-    }.bind(this));
+    return new Promise((resolve, reject) => {
+      channels.create(current, (err, data) => {
+        this.updateCurrentData_(err, data, options);
+        if (err) {
+          reject(err);
+          return;
+        }
+        this.save(
+          false /* forceNewVersion */,
+          true /* preparingRemix */
+        ).then(resolve).catch(reject);
+      });
+    });
```

Comment:
The `.then(resolve).catch(reject)` feels like an antipattern.  You can simplify a little to `.then(resolve, reject)` since [`then` takes an optional `onReject` callback](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then), but if `save` already returns a promise maybe it's better to flatten the promise chain here?
```js
return new Promise((resolve, reject) => {
  channels.create(current, (err, data) => {
    this.updateCurrentData_(err, data, options);
    err ? reject(err) : resolve();
  });
}).then(() => this.save(
  false /* forceNewVersion */,
  true /* preparingRemix */
));
```